frontend/src/component/feature/FeatureStrategy/NewFeatureStrategyCreate/NewFeatureStrategyCreate.test.tsx

```diff
@@ -1,20 +1,322 @@
+                    environment: 'development',
+                },
+                {
+                    permission: UPDATE_FEATURE_STRATEGY,
+                    project: 'default',
+                    environment: 'development',
+                },
+                {
+                    permission: UPDATE_FEATURE_ENVIRONMENT_VARIANTS,
+                    project: 'default',
+                    environment: 'development',
+                },
+            ],
+        },
+    );
+};
+
+beforeEach(() => {
+    setupProjectEndpoint();
+    setupSegmentsEndpoint();
+    setupStrategyEndpoint();
+    setupFeaturesEndpoint();
+    setupUiConfigEndpoint();
+    setupContextEndpoint();
+});
+
+describe('NewFeatureStrategyCreate', () => {
+    test('formatAddStrategyApiCode', () => {
+        expect(
+            formatAddStrategyApiCode(
+                'projectId',
+                'featureId',
+                'environmentId',
+                { id: 'strategyId' },
+                'unleashUrl',
+            ),
+        ).toMatchInlineSnapshot(`
       "curl --location --request POST 'unleashUrl/api/admin/projects/projectId/features/featureId/environments/environmentId/strategies' \\\\
           --header 'Authorization: INSERT_API_KEY' \\\\
           --header 'Content-Type: application/json' \\\\
           --data-raw '{
         \\"id\\": \\"strategyId\\"
       }'"
     `);
+    });
+
+    test('should navigate tabs', async () => {
+        setupComponent();
+
+        await waitFor(() => {
+            expect(screen.getByText('Gradual rollout')).toBeInTheDocument();
+        });
+
+        const slider = await screen.findByRole('slider', { name: /rollout/i });
+        expect(slider).toHaveValue('100');
+
+        const targetingEl = screen.getByText('Targeting');
+        fireEvent.click(targetingEl);
+
+        await waitFor(() => {
+            expect(screen.getByText('Segments')).toBeInTheDocument();
+        });
+
+        const variantEl = screen.getByText('Variants');
+        fireEvent.click(variantEl);
+
+        await waitFor(() => {
+            expect(screen.getByText('Add variant')).toBeInTheDocument();
+        });
+    });
+
+    test('should change general settings', async () => {
+        setupComponent();
+
+        await waitFor(() => {
+            expect(screen.getByText('Gradual rollout')).toBeInTheDocument();
+        });
+
+        const slider = await screen.findByRole('slider', { name: /rollout/i });
+        const groupIdInput = await screen.getByLabelText('groupId');
+
+        expect(slider).toHaveValue('100');
+        expect(groupIdInput).toHaveValue(featureName);
+
+        fireEvent.change(slider, { target: { value: '50' } });
+        fireEvent.change(groupIdInput, { target: { value: 'newGroupId' } });
+
+        expect(slider).toHaveValue('50');
+        expect(groupIdInput).toHaveValue('newGroupId');
+    });
+
+    test('should change targeting settings', async () => {
+        setupComponent();
```

Comment:
I'd parametrize setupComponent with data that matters later in the test e.g. segmentName. Otherwise I need to read the entire setupComponent to find why we're looking for 'test' on line 295
I started exploring this testing pattern e.g. in FeatureToggleListTable.test.tsx